src/sentry/notifications/platform/rollout.py

```diff
@@ -0,0 +1,81 @@
+import logging
+from dataclasses import dataclass
+
+from sentry import features, options
+from sentry.models.organization import Organization
+from sentry.organizations.services.organization.model import RpcOrganization
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class NotificationRolloutService:
+    """
+    Holds all checks and logic for rolling out a new notification to the platform.
+      - Checks if the organization has access to the platform via rollout stage Feature Flags
+      - Gets the rollout rate for the source from the option mapped to the rollout stage
+    """
+
+    organization: RpcOrganization | Organization
+
+    def should_notify(self, source: str) -> bool:
+        option_key = self.has_feature_flag_access()
+        if option_key is None:
+            return False
+
+        source_rollout_rate = self.get_rollout_rate(option_key, source)
+        return (self.organization.id % 100) < 100 * source_rollout_rate
+
+    def has_feature_flag_access(self) -> str | None:
+        internal_testing = features.has(
+            "organizations:notification-platform.internal-testing", self.organization
+        )
+        is_sentry = features.has("organizations:notification-platform.is-sentry", self.organization)
+        early_adopter = features.has(
+            "organizations:notification-platform.early-adopter", self.organization
+        )
+        general_access = features.has(
+            "organizations:notification-platform.general-access", self.organization
+        )
+
+        option_key = None
+        if internal_testing:
+            option_key = "notifications.platform-rollout.internal-testing"
+        elif is_sentry:
+            option_key = "notifications.platform-rollout.is-sentry"
```

Comment:
Do we need a separate `is-sentry` rollout if we have the `internal-testing` one already?